src/assets/RestApi/rest-api.md

```diff
@@ -0,0 +1,268 @@
+```
+Example:
+```bash
+curl http://127.0.0.1:5000/tasks/1
+```
+Expected Result:
+```json
+{
+  "task": {
+    "completed": false,
+    "id": 1,
+    "title": "Buy groceries"
+  }
+}
+```
+
+### Create a New Task
+To create a new task, you can use following curl command, replacing `<task_data>` with the JSON data for the new task:
+```bash
+curl -X POST -H "Content-Type: application/json" -d '<json_data>' http://127.0.0.1:5000/tasks
+```
+Example:
+```bash
+curl -X POST -H "Content-Type: application/json" -d '{"id":3,"task":"Finish the report","completed":false}' http://127.0.0.1:5000/tasks
+```
+Expected Result:
+```json
+{
+  "task": {
+    "completed": false,
+    "id": 3,
+    "task": "Finish the report"
+  }
+}
+```
+
+### Update a task
+To update an existing task, use the following curl command, replacing `<task_id>` with the id of the task, and `<updated_data>` with the JSON data you want to update:
+```bash
+curl -X PUT -H "Content-Type: application/json" -d '<updated_data>' http://127.0.0.1:5000/tasks/<task_id>
+```
+Example:
+```bash
+curl -X PUT -H "Content-Type: application/json" -d '{"completed":true}' http://127.0.0.1:5000/tasks/3
+```
+Expected Result:
+```json
+{
+  "task": {
+    "completed": true,
+    "id": 3,
+    "task": "Finish the report"
+  }
+}
+```
+
+### Delete a task
+To delete a task, use the following curl command, replacing `<task_id>` with the id of the task you want to delete:
+```bash
+curl -X DELETE http://127.0.0.1:5000/tasks/<task_id>
+```
+Example:
+```bash
+curl -X DELETE http://127.0.0.1:5000/tasks/3
+```
+Expected Result:
+```json
+{
+  "result": "Task deleted"
+}
+```
+
+Let's try to retrieve a task which doesn't exist in the list of task. For instance, we just deleted task with id = 3. Now, let's try to retrieve the task with id = 3 and see the result.
+```bash
+curl http://127.0.0.1:5000/tasks/3
+```
+Expected result:
+```json
+{
+  "error": "Task not found"
+}
+```
+We see that the JSON response returns the error `Task not found`, which indicates that the task doesn't exist in the list of tasks.
+
+Now, let's look into the cases with invalid requests:
+1. **Invalid endpoint**:
+Try to get all the tasks with following curl command:
+```bash
+curl http://127.0.0.1:5000/task
+```
+Here, we have used `task` instead of `tasks` in our endpoint, which makes our endpoint invalid. With this request we get the following error with `404` status code:
+```console
+The requested URL was not found on the server. If you entered the URL manually please check your spelling and try again.
+```
+
+2. **Invalid JSON data**:
+Try to add a new task with invalid JSON data:
+```bash
+curl -X POST -H "Content-Type: application/json" -d '{"id":3,task:"Finish the report","completed":false}' http://127.0.0.1:5000/tasks
+```
+Here, the JSON data in request body is invalid as the key `task` is not enclosed in double quotes. With this request we get the following error with status code `400`.
```

Comment:
Before going into specific examples of invalid data, maybe mention some of the most common syntax and structure errors...